### PR TITLE
Add skip preflight to CLI

### PR DESCRIPTION
**Pull request by Woody4618**
#### Problem
CLI was missing the option to set skip-preflight option. 
This is especially useful because some RPC providers are not using swqos when preflight is enabled. 
This is speeding up program deployments by a lot using these RPC providers.  

#### Summary of Changes
I added skip-preflight as global flag and changed all the places where transactions are sent to use that flag. 
Some test had the skip preflight flag set to true, but it was not actually used. So these tests that were parsing the responses are adjusted to handle both cases. 

#### Refactoring? 
In an additional step if would maybe be nice to remove CommitmentConfig from the CliConfig and move it into the RpcSendTransactionConfig. But that looked like a bit of a bigger refactoring so I didnt include it. 
Another option would also be moving RpcSendTransactionConfig directly into RpcClient to enable it everywhere. But that is an even bigger refactoring.

### CODE REVIEW

File: cargo-registry/src/client.rs
Comment (joncinque) [inline cargo-registry/src/client.rs:221]: If you want to make the change in here too, `skip_preflight` needs to be declared higher up as an option in `get_clap_app`.

But also, we'll need to pass `rpc_transaction_config` to `process_program_deploy` if we want it to respect skip-preflight.

File: cli/src/clap_app.rs
Comment (joncinque) [inline cli/src/clap_app.rs]: nit: since this is a flag (ie. an arg that doesn't take a value) there's no need to specify a default
```suggestion
                .help("Skip the preflight check when sending transactions"),
```

File: rpc-client/src/nonblocking/rpc_client.rs
Comment (joncinque) [inline rpc-client/src/nonblocking/rpc_client.rs]: This line is redundant -- the `..config` two lines later ensures that it'll use whatever is already in `config`. Otherwise, this would have never worked :sweat_smile:

File: client/src/send_and_confirm_transactions_in_parallel.rs
Comment (joncinque) [inline client/src/send_and_confirm_transactions_in_parallel.rs]: This is a breaking change. This means that we'll need to create a new struct `SendAndConfirmConfigV2`, and create new versions of `send_and_confirm_transactions_in_parallel_blocking` / `send_and_confirm_transactions_in_parallel` (probably just add `_v2` to the end), deprecate the old functions and struct, and have the deprecated functions call into the new ones with a default for the new field. Does that make sense?

And while we're at it, why not just add the whole `RpcSendTransactionConfig`?

File: cli/tests/transfer.rs
Comment (joncinque) [inline cli/tests/transfer.rs]: While the test coverage is nice, since these tests are pretty slow, let's revert this and just do it in the previous test

File: cli/src/program_v4.rs
Comment (joncinque) [inline cli/src/program_v4.rs:1120]: This is a change in behavior since it used to always skip preflight, but since it's only 1 or 2 transactions, I think it's fine to respect the user's preflight choice. It won't slow things down too much more

File: cli/src/program.rs
Comment (joncinque) [inline cli/src/program.rs:1732]: I can't comment higher up, but there's a `send_and_confirm_transaction_with_spinner` call on line 1578 that isn't using the send transaction config -- could you add it there?

File: cli/tests/program.rs
Comment (joncinque) [inline cli/tests/program.rs]: For the sake of faster tests, we can probably remove the test-case here, since the interesting behavior is for the program deployment, and it's covered elsewhere

**Review by joncinque — COMMENTED** [review]
Really good work overall, thanks for the contribution! Just a few things to fixup, then we can land this

File: cli/src/program_v4.rs
Comment (Woody4618) [reply cli/src/program_v4.rs:1120]: Oh you mean if someone is calling it with skip_preflight true and before it was ignored and now its used? Should i change something about it?

File: cli/src/program.rs
Comment (Woody4618) [reply cli/src/program.rs:1732]: Oh thanks! I missed that

**Comment by Woody4618** [comment]
Thank you very much for the review! Pushed the changes :)

File: cli/src/program_v4.rs
Comment (joncinque) [reply cli/src/program_v4.rs:1120]: Nah, I think it's fine

File: client/src/send_and_confirm_transactions_in_parallel.rs
Comment (joncinque) [inline client/src/send_and_confirm_transactions_in_parallel.rs]: nit: can you add a "since" field?
```suggestion
#[deprecated(since = "2.2.0", note = "Use send_and_confirm_transactions_in_parallel_blocking_v2")]
```

File: client/src/send_and_confirm_transactions_in_parallel.rs
Comment (joncinque) [inline client/src/send_and_confirm_transactions_in_parallel.rs]: nit: since field plz
```suggestion
#[deprecated(since = "2.2.0", note = "Use send_and_confirm_transactions_in_parallel_v2")]
```

File: client/src/send_and_confirm_transactions_in_parallel.rs
Comment (joncinque) [inline client/src/send_and_confirm_transactions_in_parallel.rs]: nit: since here too
```suggestion
#[deprecated(since = "2.2.0", note = "Use SendAndConfirmConfigV2 with send_and_confirm_transactions_in_parallel_v2")]
```

File: client/src/send_and_confirm_transactions_in_parallel.rs
Comment (joncinque) [inline client/src/send_and_confirm_transactions_in_parallel.rs]: nit: can just go straight with default here
```suggestion
        rpc_send_transaction_config: RpcSendTransactionConfig::default(),
```

File: client/src/send_and_confirm_transactions_in_parallel.rs
Comment (joncinque) [inline client/src/send_and_confirm_transactions_in_parallel.rs]: nit: can just use default directly:
```suggestion
        rpc_send_transaction_config: RpcSendTransactionConfig::default(),
```

**Review by joncinque — COMMENTED** [review]
Just a couple of little nits, then we can get this in!

**Comment by joncinque** [comment]
Be sure to run `./cargo nightly fmt --all` from the top of the repo, looks like there are some formatting issues. You can also run `./scripts/cargo-clippy-nightly.sh` for any lint issues.

**Comment by Woody4618** [comment]
It looks like the linter does not want me to use deprecated functions. But since its the fallback. How can i not use them? 
image

**Comment by joncinque** [comment]
> It looks like the linter does not want me to use deprecated functions. But since its the fallback. How can i not use them?

You can add `#[allow(deprecated)]` for the deprecated functions

**Comment by Woody4618** [comment]
Can i Ignore this? when running ./scripts/cargo-clippy-nightly.sh 

image

**Comment by joncinque** [comment]
> Can i Ignore this? when running ./scripts/cargo-clippy-nightly.sh

You can always push your change and see if it passes CI, but it's better to run things locally. Are you missing a protobuf compiler on your machine?

**Comment by Woody4618** [comment]
I have it installed yes, via homebrew. 
protoc --version                                                                           
libprotoc 28.3 
Looks like the linting worked though.

**Comment by joncinque** [comment]
Ah darn, sorry, I forgot to mention -- can you add an entry in the changelog for this feature? Under 2.2.0, how about something like:

```
  * Changes
    * CLI:
      * Add global `--skip-preflight` option for skipping preflight checks on all transactions sent through RPC. This flag, along with `--use-rpc`, can improve success rate with program deployments using the public RPC nodes.
```

**Comment by Woody4618** [comment]
Added the changelog entry

**Comment by joncinque** [comment]
> Added the changelog entry

Might need a rebase -- there's a conflict

**Comment by joncinque** [comment]
@Woody4618 please double-check it, but this should be correct

**Review by joncinque — APPROVED** [review]
Thanks for your contribution!

**Comment by joncinque** [comment]
@febo can I get an approval here? I cleaned up the branch after a rebase, and it won't let me merge since I was the last pusher

**Review by febo — APPROVED** [review]

**Pull request merged**
